include/aspect/heating_model/interface.h

```diff
@@ -79,6 +80,21 @@ namespace aspect
         update ();
 
         /**
+         * Function to compute the heating terms in @p heating_outputs given the
+         * inputs in @p material_model_inputs and the outputs of the material
+         * model in @p material_model_outputs. The @p heating_outputs vector is
+         * filled with the value of the heating rate at each quadrature point as
+         * defined in @p material_model_inputs.
+         * The default implementation calls specific_heating_rate to make
+         * this implementation backwards compatible.
+         */
+        virtual
+        void
+        execute (const typename aspect::MaterialModel::Interface<dim>::MaterialModelInputs &material_model_inputs,
+                 const typename aspect::MaterialModel::Interface<dim>::MaterialModelOutputs &material_model_outputs,
+                 std::vector<double> &heating_outputs) const;
```

Comment:
A function called `execute` typically does some action, but in common use of the word does not return anything. How about renaming it to `compute_specific_heating_rates` or similar?